### PR TITLE
feat: added meal prerendering ( #36 )

### LINKED ISSUE #36: 급식 사전 렌더링
급식 사진 데이터를 이용하여, 사전 렌더링

### PR DESCRIPTION
**Pull request by wonderlandpark**
#36 구현

### CODE REVIEW

File: schoolbot/cogs/events/error.py [hunk 1]
```diff
@@ -63,5 +63,5 @@ async def on_command_error(self, ctx, error):
                     name="Traceback:",
                     value=f"```py\n{''.join(traceback.format_exception(type(error.__cause__), error.__cause__, error.__cause__.__traceback__, limit=3))}\n```",
                 )
-            channel = await self.bot.fetch_channel(os.environ["channel_id"])
+            channel = await self.bot.fetch_channel('745844596176715806')
```
Comment (kijk2869) [inline schoolbot/cogs/events/error.py]:
```suggestion
            channel = await self.bot.fetch_channel(os.environ["channel_id"])
```

**Review by kijk2869 — CHANGES_REQUESTED** [review]

**Review by SaidBySolo — DISMISSED** [review]
image

how to solve like this case?

**Comment by kijk2869** [comment]
> image
> 
> how to solve like this case?

https://github.com/callistoteam/mealAPI/issues/2

**Review by kijk2869 — APPROVED** [review]

File: schoolbot/cogs/school/meal.py [hunk 2]
```diff
@@ -4,13 +4,20 @@
...
 
 class Meal(commands.Cog):
     def __init__(self, bot, apikey):
         self.bot = bot
         self.neis = neispy.AsyncClient(apikey)
 
+    async def render_meal_image(self, meals):
+        async with aiohttp.ClientSession() as session:
+            meals = '&meal='.join(meals.split('<br/>'))
+            async with session.post(f"https://api.schoolbot.callisto.team/render/?meal={meals}", headers={ "Authorization":  "youshallnotpass" }) as response:
```
Comment (SaidBySolo) [inline schoolbot/cogs/school/meal.py]: 토큰 하드코딩에 유출까지 해내셨습니다.

**Review by kijk2869 — CHANGES_REQUESTED** [review]
Should lint with black

File: schoolbot/cogs/school/meal.py (hunk 2 above)
Comment (SaidBySolo) [inline schoolbot/cogs/school/meal.py]:
```suggestion
import os
from neispy import DataNotFound
import discord
from discord.ext import commands
import aiohttp


class Meal(commands.Cog):
    def __init__(self, bot, apikey):
        self.bot = bot
        self.neis = neispy.AsyncClient(apikey)

    async def render_meal_image(self, meals):
        async with aiohttp.ClientSession() as session:
            meals = '&meal='.join(meals.split('<br/>'))
            async with session.post(f"https://api.schoolbot.callisto.team/render/?meal={meals}", headers={ "Authorization":  os.environ["MEAL_API_KEY"] }) as response:
```

**Review by SaidBySolo — CHANGES_REQUESTED** [review]
원더님 미치셨나요

File: schoolbot/cogs/school/meal.py (hunk 2 above)
Comment (kijk2869) [reply schoolbot/cogs/school/meal.py]: 근데 이부분은 어케할 방법이 없음

Comment (kijk2869) [reply schoolbot/cogs/school/meal.py]: Auth 유출은 그렇다 치는데

Comment (SaidBySolo) [reply schoolbot/cogs/school/meal.py]: ㄴㄴ 토큰하드코딩 때문에 그런거

File: schoolbot/cogs/school/meal.py [hunk 3]
```diff
@@ -4,13 +4,20 @@
...
 
 
 class Meal(commands.Cog):
     def __init__(self, bot, apikey):
         self.bot = bot
         self.neis = neispy.AsyncClient(apikey)
 
+    async def render_meal_image(self, meals):
+        async with aiohttp.ClientSession() as session:
+            meals = '&meal='.join(meals.split('<br/>'))
```
Comment (kijk2869) [inline schoolbot/cogs/school/meal.py]: need url encoding

**Review by kijk2869 — CHANGES_REQUESTED** [review]

**Comment by wonderlandpark** [comment]
Assigned @SaidBySolo for this problem

**Comment by wonderlandpark** [comment]
I'm Busy

**Comment by wonderlandpark** [comment]
image

**Comment by kijk2869** [comment]
@SaidBySolo is reviewing #49

**Comment by wonderlandpark** [comment]
Resolved Every Problem

**Comment by wonderlandpark** [comment]
Requesting to **Experienced Python developer** @SaidBySolo for formating code by style rule.

File: schoolbot/cogs/school/meal.py [hunk 4]
```diff
@@ -1,16 +1,24 @@
...
```
Comment (SaidBySolo) [inline schoolbot/cogs/school/meal.py]: You not resolve this conversation
i think you push in another branch
please check

**Review by SaidBySolo — CHANGES_REQUESTED** [review]

**Comment by wonderlandpark** [comment]
@SaidBySolo Please rereview

File: schoolbot/cogs/school/meal.py [hunk 5]
```diff
@@ -1,16 +1,26 @@
...
```
Comment (SaidBySolo) [inline schoolbot/cogs/school/meal.py]: import urllib.parse but not used

**Review by SaidBySolo — CHANGES_REQUESTED** [review]

File: schoolbot/cogs/school/meal.py [hunk 6]
```diff
@@ -1,16 +1,26 @@
 import asyncio
+import os
 
 import neispy
+import os
```
Comment (SaidBySolo) [inline schoolbot/cogs/school/meal.py]: why twice import os?

**Review by SaidBySolo — CHANGES_REQUESTED** [review]

**Review by kijk2869 — CHANGES_REQUESTED** [review]
image

File: schoolbot/cogs/school/meal.py (hunk 6 above)
Comment (wonderlandpark) [reply schoolbot/cogs/school/meal.py]: Outdated Please Check the updated commit

**Review by kijk2869 — APPROVED** [review]

**Review by SaidBySolo — APPROVED** [review]

**Pull request merged**